### PR TITLE
Fix libiconv support

### PR DESCRIPTION
**Pull request by neheb**
Autotools provide AM_ICONV for this.

some libiconv implementations use const char for iconv. Check for it.

### CODE REVIEW

**Comment by 2franix** [comment]
Thanks for pointing that out. Can you please rework the description of the PR to make it clearer for people not familiar with autotools?
I'll try to have at look at the build failure.

**Comment by neheb** [comment]
Not sure what you wanted but here.

**Comment by 2franix** [comment]
I would like the PR to describe the problem it is trying to fix. "Fix libiconv support" is not self explanatory. In case your fix causes an issue in the future, it would be nice to have a well designed comment to help understand the problem.

**Comment by neheb** [comment]
Maybe this?

**Comment by neheb** [comment]
Turns out I had one ICONV_CONST too many. Updated.

**Comment by 2franix** [comment]
To fix the broken build, I suggest you try the fix suggested by [this post](https://stackoverflow.com/a/11683882). This is certainly doable by modifying the command lines for `autoreconf` and probably `configure` in the "script:" section of .travis.yml.

But as a sidenote, let me confess I am a bit confused by the fact that in spite of this automake macro to support iconv, the iconv package does not seem to be properly handled without specifying additional paths when building. Even `pkg-config --list-all` does not find any iconv package. I am definitely not an expert with the autotools, though.
In your opinion, what is the added value of using AM_ICONV in place of the config we currently have in master?

**Comment by neheb** [comment]
It handles ICONV_CONST. Without it, there is a compilation error.

**Comment by 2franix** [comment]
OK, then try to fix the build as per my suggestion and let us know. Thanks!

**Comment by neheb** [comment]
No idea about this linking error.

**Comment by 2franix** [comment]
The link issue seems related to [this issue](https://stackoverflow.com/a/409402). But this is weird because `-lcppunit` almost comes last in the linker command line.
The most accepted answer gives a [practical example](https://stackoverflow.com/a/409470).
Why this issue arises now and only for gcc 4.9, I don't know. But since version 4.9 is pretty old now, I suggest to get rid of this config from the build matrix. Let's say g++-4.9 is no longer officially supported.

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ matrix:
                   packages:
                       - g++-6
                       - libcppunit-dev
+                      - gettext
```
Comment (2franix) [inline .travis.yml]: @neheb  Please rework your commit comment - or at least the description of the PR - to explain **why** you add gettext. Your current commit comment only paraphrase the code without giving a reason. Thanks!

**Comment by neheb** [comment]
How about now?

**Comment by 2franix** [comment]
Looks good now, thanks for the update!

**Pull request merged**